docs/tokencredentials.md

```diff
@@ -0,0 +1,163 @@
+{
+    ClientId = clientId,
+};
+DeviceCodeCredential deviceCodeCredential = new DeviceCodeCredential(deviceCodeCredentialOptions);
+
+GraphServiceClient graphServiceClient = new GraphServiceClient(deviceCodeCredential, scopes);
+
+User me = await graphClient.Me.Request()
+                .GetAsync();
+```
+
+## 4. ClientSecretCredential
+
+This credential class provides a similar use to the [Client Credential Provider](https://github.com/microsoftgraph/msgraph-sdk-dotnet-auth#b-client-credential-provider) with the use of a client secret and can be used as follows.
+The device code flow enables sign in to devices by way of another device.
+
+```cs
+string[] scopes = {"User.Read"};
+
+ClientSecretCredential clientSecretCredential = new ClientSecretCredential(tenantId, clientId, clientSecret); 
+
+GraphServiceClient graphServiceClient = new GraphServiceClient(clientSecretCredential, scopes);
+
+User me = await graphClient.Me.Request()
+                .GetAsync();
+```
+
+## 5. ClientCertificateCredential
+
+This credential class provides a similar use to the [Client Credential Provider](https://github.com/microsoftgraph/msgraph-sdk-dotnet-auth#b-client-credential-provider) with the use of a client certificate and can be used as follows.
+
+```cs
+string[] scopes = {"User.Read"};
+
+ClientCertificateCredential clientCertificateCredential = new ClientCertificateCredential(tenantId, clientId, certificatePath);
+
+// or pass instance of X509Certificate2
+// ClientCertificateCredential clientCertificateCredential = new ClientCertificateCredential(tenantId, clientId, certificatePath);
+
+GraphServiceClient graphServiceClient = new GraphServiceClient(clientCertificateCredential, scopes);
+
+User me = await graphClient.Me.Request()
+                .GetAsync();
+```
+
+## 6. AuthorizationCodeCredential
+
+This credential class provides a similar use to the [Authorization code provider](https://github.com/microsoftgraph/msgraph-sdk-dotnet-auth#a-authorization-code-provider) with the use of a client secret and can be used as follows. The authorization code flow enables native and web apps to securely obtain tokens in the name of the user. 
+
+```cs
+string[] scopes = {"User.Read"};
+
+AuthorizationCodeCredential authorizationCodeCredential = new AuthorizationCodeCredential(tenantId, clientId,  clientSecret, authCode);
+
+GraphServiceClient graphServiceClient = new GraphServiceClient(authorizationCodeCredential, scopes);
+
+User me = await graphClient.Me.Request()
+                .GetAsync();
+```
+
+# Other TokenCredentials
+
+It is also useful to know that Azure.Identity provides other TokenCredentials that may be useful. These include.
+
+## 1. EnvironmentCredential
+
+This credential enables the used of defined environment variables to configure authentication. The environment varables used are as follows.
+
+AZURE_TENANT_ID -   The Azure Active Directory tenant(directory) ID.
```

Comment:
nit: format as list?

Reply:
Thanks, will action and resolve

Reply:
Resolved via https://github.com/microsoftgraph/msgraph-sdk-dotnet/pull/942/commits/65059b5001dd2b17e327ff43a23144eef2f40d65